---
title: "Gene Expression Analysis of Rat Testis Across Developmental Stages"
author: "Brad Hansen"
date: "3/24/2022"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# **Gene Expression Analysis of Rat Testis Across Developmental Stages**

### Data from [NCBI GEO](https://www.ncbi.nlm.nih.gov/geo/)

Data used:

> GEO study [GSE108348](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE108348)
>
>> Study GSE108348 includes transcriptomes from three species (*mouse, rat, chicken*), four organs (*testis, brain, liver, kidney*), across five developmental stages (*E 13.5, E 18.5-19, PND 1-2, 8-10 weeks, 24 months*). This project considers the testis data for rats (Wistar) across the development stages. The data is Bulk RNA-seq from a Illumina HiSeq 2000.

> GEO study XXXYYYZZZ
>
>> Study XXXYYYZZZ includes...

> GEO study XXXYYYZZZ
>
>> Study XXXYYYZZZ includes...

> GEO study XXXYYYZZZ
>
>> Study XXXYYYZZZ includes...

> GEO study XXXYYYZZZ\
>
>> Study XXXYYYZZZ includes...





```{r load packages, include=FALSE}
if (!require("pacman"))install.packages("pacman")
pacman::p_load(DESeq2, plyranges, tidyverse, Glimma, dplyr)



```

## Count Matrix

Generate a counts matrix with genes as the rows and samples as the column.

```{r rat data}
counts <- list.files('data/', pattern = '.txt.gz$', full.names = T) %>% 
  as_tibble_col('file') %>%
  mutate(data = map(file, read.delim)) %>% 
  mutate(sample_name = str_extract(file, '[^/]+(?=.txt.gz$)')) %>% 
  select(-file) %>%
  unnest(cols = c(data))%>%
  column_to_rownames("GeneID")

## subset to testis only

testis_counts <- counts[33:40]
```

## Metadata

Add sample information

```{r metadata, echo=FALSE}
metdata <- colnames(counts[2:length(counts)-1]) %>%
  as_tibble_col("sample_name")%>%
  add_column(tissue= NA, rep = NA, time=NA)  


metdata$rep <- str_sub(metdata$sample_name,-1)
metdata$tissue <- str_extract(metdata$sample_name, "Kidney|Testis|Liver|Brain")
metdata$time <- str_extract(metdata$sample_name, "LateEmbryo|EarlyEmbryo|Newborn|Adult|Aged")

#subset to testis only
metdata_testis <- metdata[metdata$tissue=="Testis",]%>% column_to_rownames("sample_name")
```

## Differential Expression with DESeq2

The subsetted objects for testis are used in DESeq2 with time as the design factor

```{r diff exp}

ddsObject <- DESeq2::DESeqDataSetFromMatrix(countData = testis_counts, colData = metdata_testis, design = ~ time)
dds_testis <- DESeq2::DESeq(ddsObject)

```
```{r comparison}

lfc <- list(sample1 = c("Adult","Newborn", "Adult" ), sample2 = c("Newborn", "LateEmbryo", "LateEmbryo")) %>% 
  as_tibble() %>%
  mutate(deseq_results = map2(sample1, sample2, function(x, y)
    DESeq2::results(dds_testis, contrast = c("time", x, y)))) %>%
  mutate(lfc = map(deseq_results, function(res)
    res[c('baseMean', 'log2FoldChange', 'pvalue')] %>% as.data.frame() %>% rownames_to_column("GeneID"))) %>%
  select(-deseq_results) %>%
  unnest() %>%
  rename(mean_counts = baseMean, lfc = log2FoldChange) %>%
  arrange(lfc) %>%
  print()

```

## Try Glimma

```{r glimma}

#glimmaMDS(dds_testis)

glimmaMA(dds_testis)


```

```{r biomart}

library(biomaRt)

ensembl <- useEnsembl(biomart = "genes", dataset = "rnorvegicus_gene_ensembl")

searchAttributes(mart = ensembl, pattern = "name")


genes <- lfc$GeneID

lfc_names <- lfc
names(lfc_names)[3] <- 'ensembl_gene_id'


G_list <- getBM(filters= "ensembl_gene_id", attributes= c("ensembl_gene_id","external_gene_name"),values=genes,mart= ensembl)

```

Add gene names back to lfc object

```{r add gene name}

testis_genenames <- dplyr::left_join(lfc_names, G_list, by='ensembl_gene_id')

testis_genenames %>% as_tibble() %>% drop_na() %>% filter(pvalue <0.05)
```

