{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Gene Expression Analysis of Rat Testis Across Developmental Stages\n",
    "\n",
    "## Brad Hansen\n",
    "\n",
    "### Data from [NCBI GEO](https://www.ncbi.nlm.nih.gov/geo/)\n",
    "\n",
    "Search terms: *testis*, *rat*, *testes*, *postnatal*,\n",
    "\n",
    "Only RNA-seq data used (not microarray).\n",
    "\n",
    "Data used:\n",
    "\n",
    "GEO study [GSE108348](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE108348)\n",
    "\n",
    "> Study GSE108348 includes transcriptomes from three species (*mouse, rat, chicken*), four organs (*testis, brain, liver, kidney*), across five developmental stages (*E 13.5, E 18.5-19, PND 1-2, 8-10 weeks, 24 months*). This project considers the testis data for rats (Wistar) across the development stages. The data is Bulk RNA-seq from a Illumina HiSeq 2000.\n",
    "\n",
    "GEO study [GSE162152](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE162152)\n",
    "\n",
    "> Study GSE162152 includes gene expression profiling of testis (*also liver and cerebellum*) across five species (*opossum, mouse, rat, rhesus macaque, human*). The authors specifically targeted circRNAs through RNA R treatment, though this analysis only uses the untreated samples. The data is Bulk RNA-seq from Illumina HiSeq 2500. Rat samples taken at 16 weeks of age.\n",
    "\n",
    "GEO study [GSE125483](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE125483)\n",
    "\n",
    "> Study GSE1235483 includes gene expression profiles from 12 tissues, across 4 species (*cynomolgus macaque, mouse, rat, and dog*). Here we use testis data from 9 week old brown norway rats.\n",
    "\n",
    "GEO study [GSE85420](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE85420)\n",
    "\n",
    "> Study GSE85420 includes gene expression profiling after exposure to 2,2',4'4' --Tetrabromodiphenyl Ether. This study looks at the testis RNA-seq data for the control group of Wistar rats at PND 120 (\\~17 weeks)\n",
    "\n",
    "GEO study [GSE41637](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE41637)\n",
    "\n",
    "> Study GSE41637 looks at transciptome differences across mammals. The study reports that samples taken from animals *of breeding age* due to transcriptome stability. Samples analyses (for Rattus norvegicus) using Illumina Genome Analyzer IIx.\n",
    "\n",
    "GEO study [GSE53960](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE53960)\n",
    "\n",
    "> Study GSE53960 assesses the *\"transcriptomic profiling across 11 organs, 4 ages, and 2 sexes of Fischer 344 rats.\"* This study uses the testis data from 2, 6, 21, and 104 week old Fischer 344 rats.\n",
    "\n",
    "\n",
    "# Periods Covered\n",
    "\n",
    "\n",
    "| Period     | Species    | Strain      | Source                        |\n",
    "|-------|-----|-----|----|\n",
    "| Embryonic Day 13.5 (*midstage embryo*)      | Rat        | Wistar                            | [GSE108348](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE108348)                                                                            |\n",
    "| Embryonic Day 18.5-19 (*late Embryo*)       | Rat        | Wistar                            | [GSE108348](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE108348)                                                                            |\n",
    "| Post Natal Day 1-2 (*neonate*)              | Rat        | Wistar                            | [GSE108348](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE108348)                                                                            |\n",
    "| 2 weeks old (*young*)                       | Rat        | Fischer 344                       | [GSE53960](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE53960)                                                                              |\n",
    "| 6 weeks old (*young*)                       | Rat        | Fischer 344                       | [GSE53960](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE53960)                                                                              |\n",
    "| 8-10 weeks old (*young adult*), 9 weeks old | Rat        | Wistar, Brown rat                 | [GSE108348](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE108348), [GSE125483](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE125483) |\n",
    "| 16 weeks old (*adult*)                      | Rat        | Norway Brown (*NCBI Taxon 10116*) | [GSE162152](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE162152)                                                                            |\n",
    "| 17 weeks old (*adult*)                      | Rat        | Wistar                            | [GSE85420](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE85420)                                                                              |\n",
    "| 21 weeks old (*adult*)                      | Rat        | Fischer 344                       | [GSE53960](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE53960)                                                                              |\n",
    "| 2 years old (*aged adult*)                  | Rat        | Wistar , Fisher 344               | [GSE108348](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE108348), [GSE53960](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE53960)   |\n",
    "| *Breeding Age*                              | Rat        | Sprague-Dawley                    | [GSE41637](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE41637)                                                                              |\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: BiocGenerics\n",
      "\n",
      "\n",
      "Attaching package: ‘BiocGenerics’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    anyDuplicated, append, as.data.frame, basename, cbind, colnames,\n",
      "    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,\n",
      "    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,\n",
      "    union, unique, unsplit, which.max, which.min\n",
      "\n",
      "\n",
      "Loading required package: IRanges\n",
      "\n",
      "Loading required package: S4Vectors\n",
      "\n",
      "Loading required package: stats4\n",
      "\n",
      "\n",
      "Attaching package: ‘S4Vectors’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Loading required package: GenomicRanges\n",
      "\n",
      "Loading required package: GenomeInfoDb\n",
      "\n",
      "\n",
      "Attaching package: ‘plyranges’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:IRanges’:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:stats’:\n",
      "\n",
      "    filter\n",
      "\n",
      "\n",
      "── \u001b[1mAttaching packages\u001b[22m ─────────────────────────────────────── tidyverse 1.3.1 ──\n",
      "\n",
      "\u001b[32m✔\u001b[39m \u001b[34mggplot2\u001b[39m 3.3.5     \u001b[32m✔\u001b[39m \u001b[34mpurrr  \u001b[39m 0.3.4\n",
      "\u001b[32m✔\u001b[39m \u001b[34mtibble \u001b[39m 3.1.6     \u001b[32m✔\u001b[39m \u001b[34mdplyr  \u001b[39m 1.0.8\n",
      "\u001b[32m✔\u001b[39m \u001b[34mtidyr  \u001b[39m 1.2.0     \u001b[32m✔\u001b[39m \u001b[34mstringr\u001b[39m 1.4.0\n",
      "\u001b[32m✔\u001b[39m \u001b[34mreadr  \u001b[39m 2.1.2     \u001b[32m✔\u001b[39m \u001b[34mforcats\u001b[39m 0.5.1\n",
      "\n",
      "── \u001b[1mConflicts\u001b[22m ────────────────────────────────────────── tidyverse_conflicts() ──\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mbetween()\u001b[39m    masks \u001b[34mplyranges\u001b[39m::between()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mcollapse()\u001b[39m   masks \u001b[34mIRanges\u001b[39m::collapse()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mcombine()\u001b[39m    masks \u001b[34mBiocGenerics\u001b[39m::combine()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mdesc()\u001b[39m       masks \u001b[34mIRanges\u001b[39m::desc()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mtidyr\u001b[39m::\u001b[32mexpand()\u001b[39m     masks \u001b[34mS4Vectors\u001b[39m::expand()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mfilter()\u001b[39m     masks \u001b[34mplyranges\u001b[39m::filter(), \u001b[34mstats\u001b[39m::filter()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mfirst()\u001b[39m      masks \u001b[34mS4Vectors\u001b[39m::first()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mlag()\u001b[39m        masks \u001b[34mstats\u001b[39m::lag()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mn()\u001b[39m          masks \u001b[34mplyranges\u001b[39m::n()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mn_distinct()\u001b[39m masks \u001b[34mplyranges\u001b[39m::n_distinct()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mggplot2\u001b[39m::\u001b[32mPosition()\u001b[39m masks \u001b[34mBiocGenerics\u001b[39m::Position(), \u001b[34mbase\u001b[39m::Position()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mpurrr\u001b[39m::\u001b[32mreduce()\u001b[39m     masks \u001b[34mGenomicRanges\u001b[39m::reduce(), \u001b[34mIRanges\u001b[39m::reduce()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mrename()\u001b[39m     masks \u001b[34mS4Vectors\u001b[39m::rename()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mslice()\u001b[39m      masks \u001b[34mplyranges\u001b[39m::slice(), \u001b[34mIRanges\u001b[39m::slice()\n",
      "\n",
      "------------------------------------------------------------------------------\n",
      "\n",
      "You have loaded plyr after dplyr - this is likely to cause problems.\n",
      "If you need functions from both plyr and dplyr, please load plyr first, then dplyr:\n",
      "library(plyr); library(dplyr)\n",
      "\n",
      "------------------------------------------------------------------------------\n",
      "\n",
      "\n",
      "Attaching package: ‘plyr’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:dplyr’:\n",
      "\n",
      "    arrange, count, desc, failwith, id, mutate, rename, summarise,\n",
      "    summarize\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:purrr’:\n",
      "\n",
      "    compact\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:plyranges’:\n",
      "\n",
      "    arrange, mutate, summarise, summarize\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:IRanges’:\n",
      "\n",
      "    desc\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:S4Vectors’:\n",
      "\n",
      "    rename\n",
      "\n",
      "\n",
      "Loading required package: SummarizedExperiment\n",
      "\n",
      "Loading required package: MatrixGenerics\n",
      "\n",
      "Loading required package: matrixStats\n",
      "\n",
      "\n",
      "Attaching package: ‘matrixStats’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:plyr’:\n",
      "\n",
      "    count\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:dplyr’:\n",
      "\n",
      "    count\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘MatrixGenerics’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:matrixStats’:\n",
      "\n",
      "    colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,\n",
      "    colCounts, colCummaxs, colCummins, colCumprods, colCumsums,\n",
      "    colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,\n",
      "    colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,\n",
      "    colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,\n",
      "    colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,\n",
      "    colWeightedMeans, colWeightedMedians, colWeightedSds,\n",
      "    colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,\n",
      "    rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,\n",
      "    rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,\n",
      "    rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,\n",
      "    rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,\n",
      "    rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,\n",
      "    rowWeightedMads, rowWeightedMeans, rowWeightedMedians,\n",
      "    rowWeightedSds, rowWeightedVars\n",
      "\n",
      "\n",
      "Loading required package: Biobase\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘Biobase’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:MatrixGenerics’:\n",
      "\n",
      "    rowMedians\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:matrixStats’:\n",
      "\n",
      "    anyMissing, rowMedians\n",
      "\n",
      "\n",
      "Setting options('download.file.method.GEOquery'='auto')\n",
      "\n",
      "Setting options('GEOquery.inmemory.gpl'=FALSE)\n",
      "\n"
     ]
    }
   ],
   "source": [
    "library(plyranges)\n",
    "library(tidyverse)\n",
    "library(dplyr)\n",
    "library(plyr)\n",
    "library(DESeq2)\n",
    "library(DEGreport)\n",
    "library(GEOquery)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "GEO study [GSE108348](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE108348)\n",
    "\n",
    "Data available: text file of unique read counts \n",
    "    * Aligned with HISAT2 (2.0.5)\n",
    "    * used Ensembl release 94 annotations as a reference and we assembled additional gene models with StringTie 1.3.5.\n",
    "    * Genome build = rn6\n",
    "    * uniquely mapped read counts using the Rsubread library in R (featureCounts) version 1.32.4, in R version 3.5.\n",
    "\n",
    "Strandness of data == Illumina Truseq == use reverse (column 4 in STAR output)\n",
    "\n",
    "SRR reads:\n",
    "        SRR6396793\n",
    "        SRR6396803\n",
    "        SRR6396794\n",
    "        SRR6396795\n",
    "        SRR6396796\n",
    "        SRR6396797\n",
    "        SRR6396798\n",
    "        SRR6396799\n",
    "        SRR6396800\n",
    "        SRR6396801\n",
    "        SRR6396802"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# load in files from STAR --quantMode genecounts\n",
    "\n",
    "countPATH <- \"/bigdata/faustmanlab/bch/gse108348_counts/counts\"\n",
    "\n",
    "files <- list.files(path=countPATH, pattern=\"*.tab\", full.names=TRUE)\n",
    "names(files) <- lapply(strsplit(files, \"/|_\"), \"[\",8)\n",
    "countslist <- lapply(files, read.delim, sep=\"\\t\", header=FALSE)\n",
    "names(countslist)<- names(files)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "colnames<- c(\"gene\", \"unstranded_counts\", \"htseq_count_Fstreadalign\",\"htseq_countrev_Sndreadalign\")\n",
    "\n",
    "\n",
    "for (i in seq_along(countslist)){\n",
    "  colnames(countslist[[i]]) <- colnames}\n",
    "\n",
    "counts_temp <- lapply(countslist, function(x) {x <- x %>% select(-c(unstranded_counts,htseq_count_Fstreadalign))})\n",
    "counts_use <- lapply(counts_temp, function(x) {x <- x[-c(1:4),]})\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "all_counts <- reduce(counts_use, full_join, by = \"gene\")\n",
    "colnames(all_counts)<-c(\"gene\", names(counts_use))\n",
    "\n",
    "rownames(all_counts)<- all_counts$gene\n",
    "all_counts <- all_counts[,-1]\n",
    "\n",
    "\n",
    "\n",
    "head(all_counts)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'Run'</li><li>'Assay.Type'</li><li>'AvgSpotLen'</li><li>'Bases'</li><li>'BioProject'</li><li>'BioSample'</li><li>'Bytes'</li><li>'Center.Name'</li><li>'Consent'</li><li>'DATASTORE.filetype'</li><li>'DATASTORE.provider'</li><li>'DATASTORE.region'</li><li>'Developmental_stage'</li><li>'Experiment'</li><li>'GEO_Accession..exp.'</li><li>'Instrument'</li><li>'LibraryLayout'</li><li>'LibrarySelection'</li><li>'LibrarySource'</li><li>'Organism'</li><li>'Platform'</li><li>'ReleaseDate'</li><li>'Sample.Name'</li><li>'source_name'</li><li>'SRA.Study'</li><li>'strain'</li><li>'BREED'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'Run'\n",
       "\\item 'Assay.Type'\n",
       "\\item 'AvgSpotLen'\n",
       "\\item 'Bases'\n",
       "\\item 'BioProject'\n",
       "\\item 'BioSample'\n",
       "\\item 'Bytes'\n",
       "\\item 'Center.Name'\n",
       "\\item 'Consent'\n",
       "\\item 'DATASTORE.filetype'\n",
       "\\item 'DATASTORE.provider'\n",
       "\\item 'DATASTORE.region'\n",
       "\\item 'Developmental\\_stage'\n",
       "\\item 'Experiment'\n",
       "\\item 'GEO\\_Accession..exp.'\n",
       "\\item 'Instrument'\n",
       "\\item 'LibraryLayout'\n",
       "\\item 'LibrarySelection'\n",
       "\\item 'LibrarySource'\n",
       "\\item 'Organism'\n",
       "\\item 'Platform'\n",
       "\\item 'ReleaseDate'\n",
       "\\item 'Sample.Name'\n",
       "\\item 'source\\_name'\n",
       "\\item 'SRA.Study'\n",
       "\\item 'strain'\n",
       "\\item 'BREED'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'Run'\n",
       "2. 'Assay.Type'\n",
       "3. 'AvgSpotLen'\n",
       "4. 'Bases'\n",
       "5. 'BioProject'\n",
       "6. 'BioSample'\n",
       "7. 'Bytes'\n",
       "8. 'Center.Name'\n",
       "9. 'Consent'\n",
       "10. 'DATASTORE.filetype'\n",
       "11. 'DATASTORE.provider'\n",
       "12. 'DATASTORE.region'\n",
       "13. 'Developmental_stage'\n",
       "14. 'Experiment'\n",
       "15. 'GEO_Accession..exp.'\n",
       "16. 'Instrument'\n",
       "17. 'LibraryLayout'\n",
       "18. 'LibrarySelection'\n",
       "19. 'LibrarySource'\n",
       "20. 'Organism'\n",
       "21. 'Platform'\n",
       "22. 'ReleaseDate'\n",
       "23. 'Sample.Name'\n",
       "24. 'source_name'\n",
       "25. 'SRA.Study'\n",
       "26. 'strain'\n",
       "27. 'BREED'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"Run\"                 \"Assay.Type\"          \"AvgSpotLen\"         \n",
       " [4] \"Bases\"               \"BioProject\"          \"BioSample\"          \n",
       " [7] \"Bytes\"               \"Center.Name\"         \"Consent\"            \n",
       "[10] \"DATASTORE.filetype\"  \"DATASTORE.provider\"  \"DATASTORE.region\"   \n",
       "[13] \"Developmental_stage\" \"Experiment\"          \"GEO_Accession..exp.\"\n",
       "[16] \"Instrument\"          \"LibraryLayout\"       \"LibrarySelection\"   \n",
       "[19] \"LibrarySource\"       \"Organism\"            \"Platform\"           \n",
       "[22] \"ReleaseDate\"         \"Sample.Name\"         \"source_name\"        \n",
       "[25] \"SRA.Study\"           \"strain\"              \"BREED\"              "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 11 × 7</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Run</th><th scope=col>BioSample</th><th scope=col>Developmental_stage</th><th scope=col>Experiment</th><th scope=col>Sample.Name</th><th scope=col>strain</th><th scope=col>age_week</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>SRR6396793</td><td>SAMN08214008</td><td>young adult (8-10 weeks)    </td><td>SRX3489972</td><td>GSM2895826</td><td>Wistar</td><td>  9.00</td></tr>\n",
       "\t<tr><td>SRR6396794</td><td>SAMN08214008</td><td>young adult (8-10 weeks)    </td><td>SRX3489972</td><td>GSM2895826</td><td>Wistar</td><td>  9.00</td></tr>\n",
       "\t<tr><td>SRR6396795</td><td>SAMN08214100</td><td>young adult (8-10 weeks)    </td><td>SRX3489973</td><td>GSM2895827</td><td>Wistar</td><td>  9.00</td></tr>\n",
       "\t<tr><td>SRR6396796</td><td>SAMN08214100</td><td>young adult (8-10 weeks)    </td><td>SRX3489973</td><td>GSM2895827</td><td>Wistar</td><td>  9.00</td></tr>\n",
       "\t<tr><td>SRR6396797</td><td>SAMN08214099</td><td>aged adult (24 months)      </td><td>SRX3489974</td><td>GSM2895828</td><td>Wistar</td><td>104.00</td></tr>\n",
       "\t<tr><td>SRR6396798</td><td>SAMN08214098</td><td>aged adult (24 months)      </td><td>SRX3489975</td><td>GSM2895829</td><td>Wistar</td><td>104.00</td></tr>\n",
       "\t<tr><td>SRR6396799</td><td>SAMN08214097</td><td>embryonic day 18.5-19       </td><td>SRX3489976</td><td>GSM2895830</td><td>Wistar</td><td> -0.43</td></tr>\n",
       "\t<tr><td>SRR6396800</td><td>SAMN08214097</td><td>embryonic day 18.5-19       </td><td>SRX3489976</td><td>GSM2895830</td><td>Wistar</td><td> -0.43</td></tr>\n",
       "\t<tr><td>SRR6396801</td><td>SAMN08214096</td><td>embryonic day 18.5-19       </td><td>SRX3489977</td><td>GSM2895831</td><td>Wistar</td><td> -0.43</td></tr>\n",
       "\t<tr><td>SRR6396802</td><td>SAMN08214095</td><td>newborn (post-natal day 1-2)</td><td>SRX3489978</td><td>GSM2895832</td><td>Wistar</td><td>  0.20</td></tr>\n",
       "\t<tr><td>SRR6396803</td><td>SAMN08214094</td><td>newborn (post-natal day 1-2)</td><td>SRX3489979</td><td>GSM2895833</td><td>Wistar</td><td>  0.20</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 11 × 7\n",
       "\\begin{tabular}{lllllll}\n",
       " Run & BioSample & Developmental\\_stage & Experiment & Sample.Name & strain & age\\_week\\\\\n",
       " <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <dbl>\\\\\n",
       "\\hline\n",
       "\t SRR6396793 & SAMN08214008 & young adult (8-10 weeks)     & SRX3489972 & GSM2895826 & Wistar &   9.00\\\\\n",
       "\t SRR6396794 & SAMN08214008 & young adult (8-10 weeks)     & SRX3489972 & GSM2895826 & Wistar &   9.00\\\\\n",
       "\t SRR6396795 & SAMN08214100 & young adult (8-10 weeks)     & SRX3489973 & GSM2895827 & Wistar &   9.00\\\\\n",
       "\t SRR6396796 & SAMN08214100 & young adult (8-10 weeks)     & SRX3489973 & GSM2895827 & Wistar &   9.00\\\\\n",
       "\t SRR6396797 & SAMN08214099 & aged adult (24 months)       & SRX3489974 & GSM2895828 & Wistar & 104.00\\\\\n",
       "\t SRR6396798 & SAMN08214098 & aged adult (24 months)       & SRX3489975 & GSM2895829 & Wistar & 104.00\\\\\n",
       "\t SRR6396799 & SAMN08214097 & embryonic day 18.5-19        & SRX3489976 & GSM2895830 & Wistar &  -0.43\\\\\n",
       "\t SRR6396800 & SAMN08214097 & embryonic day 18.5-19        & SRX3489976 & GSM2895830 & Wistar &  -0.43\\\\\n",
       "\t SRR6396801 & SAMN08214096 & embryonic day 18.5-19        & SRX3489977 & GSM2895831 & Wistar &  -0.43\\\\\n",
       "\t SRR6396802 & SAMN08214095 & newborn (post-natal day 1-2) & SRX3489978 & GSM2895832 & Wistar &   0.20\\\\\n",
       "\t SRR6396803 & SAMN08214094 & newborn (post-natal day 1-2) & SRX3489979 & GSM2895833 & Wistar &   0.20\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 11 × 7\n",
       "\n",
       "| Run &lt;chr&gt; | BioSample &lt;chr&gt; | Developmental_stage &lt;chr&gt; | Experiment &lt;chr&gt; | Sample.Name &lt;chr&gt; | strain &lt;chr&gt; | age_week &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|\n",
       "| SRR6396793 | SAMN08214008 | young adult (8-10 weeks)     | SRX3489972 | GSM2895826 | Wistar |   9.00 |\n",
       "| SRR6396794 | SAMN08214008 | young adult (8-10 weeks)     | SRX3489972 | GSM2895826 | Wistar |   9.00 |\n",
       "| SRR6396795 | SAMN08214100 | young adult (8-10 weeks)     | SRX3489973 | GSM2895827 | Wistar |   9.00 |\n",
       "| SRR6396796 | SAMN08214100 | young adult (8-10 weeks)     | SRX3489973 | GSM2895827 | Wistar |   9.00 |\n",
       "| SRR6396797 | SAMN08214099 | aged adult (24 months)       | SRX3489974 | GSM2895828 | Wistar | 104.00 |\n",
       "| SRR6396798 | SAMN08214098 | aged adult (24 months)       | SRX3489975 | GSM2895829 | Wistar | 104.00 |\n",
       "| SRR6396799 | SAMN08214097 | embryonic day 18.5-19        | SRX3489976 | GSM2895830 | Wistar |  -0.43 |\n",
       "| SRR6396800 | SAMN08214097 | embryonic day 18.5-19        | SRX3489976 | GSM2895830 | Wistar |  -0.43 |\n",
       "| SRR6396801 | SAMN08214096 | embryonic day 18.5-19        | SRX3489977 | GSM2895831 | Wistar |  -0.43 |\n",
       "| SRR6396802 | SAMN08214095 | newborn (post-natal day 1-2) | SRX3489978 | GSM2895832 | Wistar |   0.20 |\n",
       "| SRR6396803 | SAMN08214094 | newborn (post-natal day 1-2) | SRX3489979 | GSM2895833 | Wistar |   0.20 |\n",
       "\n"
      ],
      "text/plain": [
       "   Run        BioSample    Developmental_stage          Experiment Sample.Name\n",
       "1  SRR6396793 SAMN08214008 young adult (8-10 weeks)     SRX3489972 GSM2895826 \n",
       "2  SRR6396794 SAMN08214008 young adult (8-10 weeks)     SRX3489972 GSM2895826 \n",
       "3  SRR6396795 SAMN08214100 young adult (8-10 weeks)     SRX3489973 GSM2895827 \n",
       "4  SRR6396796 SAMN08214100 young adult (8-10 weeks)     SRX3489973 GSM2895827 \n",
       "5  SRR6396797 SAMN08214099 aged adult (24 months)       SRX3489974 GSM2895828 \n",
       "6  SRR6396798 SAMN08214098 aged adult (24 months)       SRX3489975 GSM2895829 \n",
       "7  SRR6396799 SAMN08214097 embryonic day 18.5-19        SRX3489976 GSM2895830 \n",
       "8  SRR6396800 SAMN08214097 embryonic day 18.5-19        SRX3489976 GSM2895830 \n",
       "9  SRR6396801 SAMN08214096 embryonic day 18.5-19        SRX3489977 GSM2895831 \n",
       "10 SRR6396802 SAMN08214095 newborn (post-natal day 1-2) SRX3489978 GSM2895832 \n",
       "11 SRR6396803 SAMN08214094 newborn (post-natal day 1-2) SRX3489979 GSM2895833 \n",
       "   strain age_week\n",
       "1  Wistar   9.00  \n",
       "2  Wistar   9.00  \n",
       "3  Wistar   9.00  \n",
       "4  Wistar   9.00  \n",
       "5  Wistar 104.00  \n",
       "6  Wistar 104.00  \n",
       "7  Wistar  -0.43  \n",
       "8  Wistar  -0.43  \n",
       "9  Wistar  -0.43  \n",
       "10 Wistar   0.20  \n",
       "11 Wistar   0.20  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "\n",
    "# add metadata\n",
    "\n",
    "metadata <- read.csv(\"/bigdata/faustmanlab/bch/gse108348_counts/gse108348meta.txt\")\n",
    "\n",
    "colnames(metadata)\n",
    "\n",
    "metadata<-select(metadata,'Run', 'BioSample', 'Developmental_stage', 'Experiment', 'Sample.Name', 'strain')\n",
    "\n",
    "metadatatst <- metadata %>% \n",
    "  mutate(age_week = as.numeric(c(9,9,9,9,104,104,-.43,-.43,-.43,0.2,0.2)))\n",
    "\n",
    "metadatatst"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "dds <- DESeqDataSetFromMatrix(countData=all_counts,\n",
    "                                colData = metadata,\n",
    "                                 design = ~Developmental_stage)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "dds <- collapseReplicates(dds, groupby= dds$Sample.Name, dds$Run)\n",
    "\n",
    "dds"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "dds <- DESeq(dds)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "res <- results(dds)\n",
    "head(results(dds))\n",
    "summary(res) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "vsdata <- vst(dds, blind=FALSE)\n",
    "plotPCA(vsdata, intgroup=\"Developmental_stage\") "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "GEO study [GSE53960](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE53960)\n",
    "\n",
    "TAR of text files:\n",
    "    * \tSequenced reads were trimmed for adaptor sequence, and masked for low-complexity or low-quality sequence, then mapped to rn4 whole genome using TopHat v2.0.4 with default parameters\n",
    "    * Alignment results were then processed using Cufflinks v2.0.2 for gene and transcript quantification with default parameters. For samples with 2~3 technical replicates, average FPKM (Fragment Per Kilobase per Million mapped reads) values were used.\n",
    "    * Genome_build: rn4\n",
    "\n",
    "\n",
    "**Single -- Read length 50** [see here](https://trace.ncbi.nlm.nih.gov/Traces/sra/?run=SRR1170509)\n",
    "\n",
    "**unstranded reads -- use first column of STAR output** Method: [Illumina Tru-seq](https://www.illumina.com/products/by-type/sequencing-kits/library-prep-kits/truseq-rna-v2.html)\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "SRR reads:\n",
    "        SRR1170487\n",
    "        SRR1170488\n",
    "        SRR1170489\n",
    "        SRR1170490\n",
    "        SRR1170491\n",
    "        SRR1170492\n",
    "        SRR1170493\n",
    "        SRR1170494\n",
    "        SRR1170495\n",
    "        SRR1170496\n",
    "        SRR1170497\n",
    "        SRR1170498\n",
    "        SRR1170499\n",
    "        SRR1170500\n",
    "        SRR1170501\n",
    "        SRR1170502\n",
    "        SRR1170503\n",
    "        SRR1170504\n",
    "        SRR1170505\n",
    "        SRR1170506\n",
    "        SRR1170507\n",
    "        SRR1170508\n",
    "        SRR1170509\n",
    "        SRR1170510\n",
    "        SRR1170511\n",
    "        SRR1170512\n",
    "        SRR1170513\n",
    "        SRR1170514\n",
    "        SRR1170515\n",
    "        SRR1170516\n",
    "        SRR1170517\n",
    "        SRR1170518"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# load in files from STAR --quantMode genecounts\n",
    "\n",
    "countPATH <- \"/bigdata/faustmanlab/bch/gse53960/counts\"\n",
    "\n",
    "files <- list.files(path=countPATH, pattern=\"*.tab\", full.names=TRUE)\n",
    "names(files) <- lapply(strsplit(files, \"/|_\"), \"[\",7)\n",
    "countslist <- lapply(files, read.delim, sep=\"\\t\", header=FALSE)\n",
    "names(countslist)<- names(files)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "colnames<- c(\"gene\", \"unstranded_counts\", \"htseq_count_Fstreadalign\",\"htseq_countrev_Sndreadalign\")\n",
    "\n",
    "## select the \"unstranded_counts\" - this is specific to each study and based on the information available in NCBI Geo\n",
    "\n",
    "for (i in seq_along(countslist)){\n",
    "  colnames(countslist[[i]]) <- colnames}\n",
    "\n",
    "counts_temp <- lapply(countslist, function(x) {x <- x %>% select(-c(htseq_countrev_Sndreadalign,htseq_count_Fstreadalign))})\n",
    "counts_use <- lapply(counts_temp, function(x) {x <- x[-c(1:4),]})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "colnames<- c(\"gene\", \"unstranded_counts\", \"htseq_count_Fstreadalign\",\"htseq_countrev_Sndreadalign\")\n",
    "\n",
    "\n",
    "for (i in seq_along(countslist)){\n",
    "  colnames(countslist[[i]]) <- colnames}\n",
    "\n",
    "counts_temp <- lapply(countslist, function(x) {x <- x %>% select(-c(unstranded_counts,htseq_count_Fstreadalign))})\n",
    "counts_use <- lapply(counts_temp, function(x) {x <- x[-c(1:4),]})\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "all_counts <- reduce(counts_use, full_join, by = \"gene\")\n",
    "colnames(all_counts)<-c(\"gene\", names(counts_use))\n",
    "\n",
    "rownames(all_counts)<- all_counts$gene\n",
    "all_counts <- all_counts[,-1]\n",
    "\n",
    "\n",
    "\n",
    "head(all_counts)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'Run'</li><li>'Assay.Type'</li><li>'AvgSpotLen'</li><li>'Bases'</li><li>'BioProject'</li><li>'BioSample'</li><li>'Bytes'</li><li>'Center.Name'</li><li>'Consent'</li><li>'DATASTORE.filetype'</li><li>'DATASTORE.provider'</li><li>'DATASTORE.region'</li><li>'Experiment'</li><li>'GEO_Accession..exp.'</li><li>'Instrument'</li><li>'LibraryLayout'</li><li>'LibrarySelection'</li><li>'LibrarySource'</li><li>'Organism'</li><li>'Platform'</li><li>'ReleaseDate'</li><li>'Sample.Name'</li><li>'sex'</li><li>'source_name'</li><li>'SRA.Study'</li><li>'strain'</li><li>'tissue'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'Run'\n",
       "\\item 'Assay.Type'\n",
       "\\item 'AvgSpotLen'\n",
       "\\item 'Bases'\n",
       "\\item 'BioProject'\n",
       "\\item 'BioSample'\n",
       "\\item 'Bytes'\n",
       "\\item 'Center.Name'\n",
       "\\item 'Consent'\n",
       "\\item 'DATASTORE.filetype'\n",
       "\\item 'DATASTORE.provider'\n",
       "\\item 'DATASTORE.region'\n",
       "\\item 'Experiment'\n",
       "\\item 'GEO\\_Accession..exp.'\n",
       "\\item 'Instrument'\n",
       "\\item 'LibraryLayout'\n",
       "\\item 'LibrarySelection'\n",
       "\\item 'LibrarySource'\n",
       "\\item 'Organism'\n",
       "\\item 'Platform'\n",
       "\\item 'ReleaseDate'\n",
       "\\item 'Sample.Name'\n",
       "\\item 'sex'\n",
       "\\item 'source\\_name'\n",
       "\\item 'SRA.Study'\n",
       "\\item 'strain'\n",
       "\\item 'tissue'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'Run'\n",
       "2. 'Assay.Type'\n",
       "3. 'AvgSpotLen'\n",
       "4. 'Bases'\n",
       "5. 'BioProject'\n",
       "6. 'BioSample'\n",
       "7. 'Bytes'\n",
       "8. 'Center.Name'\n",
       "9. 'Consent'\n",
       "10. 'DATASTORE.filetype'\n",
       "11. 'DATASTORE.provider'\n",
       "12. 'DATASTORE.region'\n",
       "13. 'Experiment'\n",
       "14. 'GEO_Accession..exp.'\n",
       "15. 'Instrument'\n",
       "16. 'LibraryLayout'\n",
       "17. 'LibrarySelection'\n",
       "18. 'LibrarySource'\n",
       "19. 'Organism'\n",
       "20. 'Platform'\n",
       "21. 'ReleaseDate'\n",
       "22. 'Sample.Name'\n",
       "23. 'sex'\n",
       "24. 'source_name'\n",
       "25. 'SRA.Study'\n",
       "26. 'strain'\n",
       "27. 'tissue'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"Run\"                 \"Assay.Type\"          \"AvgSpotLen\"         \n",
       " [4] \"Bases\"               \"BioProject\"          \"BioSample\"          \n",
       " [7] \"Bytes\"               \"Center.Name\"         \"Consent\"            \n",
       "[10] \"DATASTORE.filetype\"  \"DATASTORE.provider\"  \"DATASTORE.region\"   \n",
       "[13] \"Experiment\"          \"GEO_Accession..exp.\" \"Instrument\"         \n",
       "[16] \"LibraryLayout\"       \"LibrarySelection\"    \"LibrarySource\"      \n",
       "[19] \"Organism\"            \"Platform\"            \"ReleaseDate\"        \n",
       "[22] \"Sample.Name\"         \"sex\"                 \"source_name\"        \n",
       "[25] \"SRA.Study\"           \"strain\"              \"tissue\"             "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# add metadata\n",
    "\n",
    "metadata <- read.csv(\"/bigdata/faustmanlab/bch/gse53960/gse53960_metadata.txt\")\n",
    "\n",
    "colnames(metadata)\n",
    "\n",
    "metadata<-select(metadata,'Run', 'BioSample', 'Experiment', 'Sample.Name', 'source_name', 'strain')\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "gse=getGEO(filename=\"/bigdata/faustmanlab/bch/gse53960/GSE53960_series_matrix.txt\", getGPL = FALSE)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Sample.Name</th><th scope=col>age_week</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>GSM1328469</td><td>2</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>GSM1328470</td><td>2</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>GSM1328471</td><td>2</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>GSM1328472</td><td>2</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>GSM1328473</td><td>6</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>GSM1328474</td><td>6</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 2\n",
       "\\begin{tabular}{r|ll}\n",
       "  & Sample.Name & age\\_week\\\\\n",
       "  & <chr> & <dbl>\\\\\n",
       "\\hline\n",
       "\t1 & GSM1328469 & 2\\\\\n",
       "\t2 & GSM1328470 & 2\\\\\n",
       "\t3 & GSM1328471 & 2\\\\\n",
       "\t4 & GSM1328472 & 2\\\\\n",
       "\t5 & GSM1328473 & 6\\\\\n",
       "\t6 & GSM1328474 & 6\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 2\n",
       "\n",
       "| <!--/--> | Sample.Name &lt;chr&gt; | age_week &lt;dbl&gt; |\n",
       "|---|---|---|\n",
       "| 1 | GSM1328469 | 2 |\n",
       "| 2 | GSM1328470 | 2 |\n",
       "| 3 | GSM1328471 | 2 |\n",
       "| 4 | GSM1328472 | 2 |\n",
       "| 5 | GSM1328473 | 6 |\n",
       "| 6 | GSM1328474 | 6 |\n",
       "\n"
      ],
      "text/plain": [
       "  Sample.Name age_week\n",
       "1 GSM1328469  2       \n",
       "2 GSM1328470  2       \n",
       "3 GSM1328471  2       \n",
       "4 GSM1328472  2       \n",
       "5 GSM1328473  6       \n",
       "6 GSM1328474  6       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "\n",
    "# use varData() function to check for available sample details\n",
    "\n",
    "gsemeta = data.frame(\n",
    "    names = sampleNames(gse),\n",
    "    age_week = as.numeric(gse$'developmental stage (week):ch1'))\n",
    "\n",
    "\n",
    "\n",
    "names(gsemeta)[1] <- \"Sample.Name\"\n",
    "\n",
    "head(gsemeta)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 7</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Sample.Name</th><th scope=col>Run</th><th scope=col>BioSample</th><th scope=col>Experiment</th><th scope=col>source_name</th><th scope=col>strain</th><th scope=col>age_week</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>GSM1328757</td><td>SRR1170487</td><td>SAMN02642706</td><td>SRX471656</td><td>Testes</td><td>Fisher 344</td><td>2</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>GSM1328757</td><td>SRR1170488</td><td>SAMN02642706</td><td>SRX471656</td><td>Testes</td><td>Fisher 344</td><td>2</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>GSM1328758</td><td>SRR1170489</td><td>SAMN02642705</td><td>SRX471657</td><td>Testes</td><td>Fisher 344</td><td>2</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>GSM1328758</td><td>SRR1170490</td><td>SAMN02642705</td><td>SRX471657</td><td>Testes</td><td>Fisher 344</td><td>2</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>GSM1328759</td><td>SRR1170492</td><td>SAMN02642714</td><td>SRX471658</td><td>Testes</td><td>Fisher 344</td><td>2</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>GSM1328759</td><td>SRR1170491</td><td>SAMN02642714</td><td>SRX471658</td><td>Testes</td><td>Fisher 344</td><td>2</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 7\n",
       "\\begin{tabular}{r|lllllll}\n",
       "  & Sample.Name & Run & BioSample & Experiment & source\\_name & strain & age\\_week\\\\\n",
       "  & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & GSM1328757 & SRR1170487 & SAMN02642706 & SRX471656 & Testes & Fisher 344 & 2\\\\\n",
       "\t2 & GSM1328757 & SRR1170488 & SAMN02642706 & SRX471656 & Testes & Fisher 344 & 2\\\\\n",
       "\t3 & GSM1328758 & SRR1170489 & SAMN02642705 & SRX471657 & Testes & Fisher 344 & 2\\\\\n",
       "\t4 & GSM1328758 & SRR1170490 & SAMN02642705 & SRX471657 & Testes & Fisher 344 & 2\\\\\n",
       "\t5 & GSM1328759 & SRR1170492 & SAMN02642714 & SRX471658 & Testes & Fisher 344 & 2\\\\\n",
       "\t6 & GSM1328759 & SRR1170491 & SAMN02642714 & SRX471658 & Testes & Fisher 344 & 2\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 7\n",
       "\n",
       "| <!--/--> | Sample.Name &lt;chr&gt; | Run &lt;chr&gt; | BioSample &lt;chr&gt; | Experiment &lt;chr&gt; | source_name &lt;chr&gt; | strain &lt;chr&gt; | age_week &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|\n",
       "| 1 | GSM1328757 | SRR1170487 | SAMN02642706 | SRX471656 | Testes | Fisher 344 | 2 |\n",
       "| 2 | GSM1328757 | SRR1170488 | SAMN02642706 | SRX471656 | Testes | Fisher 344 | 2 |\n",
       "| 3 | GSM1328758 | SRR1170489 | SAMN02642705 | SRX471657 | Testes | Fisher 344 | 2 |\n",
       "| 4 | GSM1328758 | SRR1170490 | SAMN02642705 | SRX471657 | Testes | Fisher 344 | 2 |\n",
       "| 5 | GSM1328759 | SRR1170492 | SAMN02642714 | SRX471658 | Testes | Fisher 344 | 2 |\n",
       "| 6 | GSM1328759 | SRR1170491 | SAMN02642714 | SRX471658 | Testes | Fisher 344 | 2 |\n",
       "\n"
      ],
      "text/plain": [
       "  Sample.Name Run        BioSample    Experiment source_name strain    \n",
       "1 GSM1328757  SRR1170487 SAMN02642706 SRX471656  Testes      Fisher 344\n",
       "2 GSM1328757  SRR1170488 SAMN02642706 SRX471656  Testes      Fisher 344\n",
       "3 GSM1328758  SRR1170489 SAMN02642705 SRX471657  Testes      Fisher 344\n",
       "4 GSM1328758  SRR1170490 SAMN02642705 SRX471657  Testes      Fisher 344\n",
       "5 GSM1328759  SRR1170492 SAMN02642714 SRX471658  Testes      Fisher 344\n",
       "6 GSM1328759  SRR1170491 SAMN02642714 SRX471658  Testes      Fisher 344\n",
       "  age_week\n",
       "1 2       \n",
       "2 2       \n",
       "3 2       \n",
       "4 2       \n",
       "5 2       \n",
       "6 2       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "all_meta = merge(x=metadata,y=gsemeta,by=\"Sample.Name\")\n",
    "\n",
    "head(all_meta)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 32 × 4</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>study</th><th scope=col>sample</th><th scope=col>strain</th><th scope=col>age_week</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>gse53960</td><td>GSM1328757</td><td>Fisher 344</td><td>2  </td></tr>\n",
       "\t<tr><td>gse53960</td><td>GSM1328757</td><td>Fisher 344</td><td>2  </td></tr>\n",
       "\t<tr><td>gse53960</td><td>GSM1328758</td><td>Fisher 344</td><td>2  </td></tr>\n",
       "\t<tr><td>gse53960</td><td>GSM1328758</td><td>Fisher 344</td><td>2  </td></tr>\n",
       "\t<tr><td>gse53960</td><td>GSM1328759</td><td>Fisher 344</td><td>2  </td></tr>\n",
       "\t<tr><td>gse53960</td><td>GSM1328759</td><td>Fisher 344</td><td>2  </td></tr>\n",
       "\t<tr><td>gse53960</td><td>GSM1328760</td><td>Fisher 344</td><td>2  </td></tr>\n",
       "\t<tr><td>gse53960</td><td>GSM1328760</td><td>Fisher 344</td><td>2  </td></tr>\n",
       "\t<tr><td>gse53960</td><td>GSM1328761</td><td>Fisher 344</td><td>6  </td></tr>\n",
       "\t<tr><td>gse53960</td><td>GSM1328761</td><td>Fisher 344</td><td>6  </td></tr>\n",
       "\t<tr><td>gse53960</td><td>GSM1328762</td><td>Fisher 344</td><td>6  </td></tr>\n",
       "\t<tr><td>gse53960</td><td>GSM1328762</td><td>Fisher 344</td><td>6  </td></tr>\n",
       "\t<tr><td>gse53960</td><td>GSM1328763</td><td>Fisher 344</td><td>6  </td></tr>\n",
       "\t<tr><td>gse53960</td><td>GSM1328763</td><td>Fisher 344</td><td>6  </td></tr>\n",
       "\t<tr><td>gse53960</td><td>GSM1328764</td><td>Fisher 344</td><td>6  </td></tr>\n",
       "\t<tr><td>gse53960</td><td>GSM1328764</td><td>Fisher 344</td><td>6  </td></tr>\n",
       "\t<tr><td>gse53960</td><td>GSM1328765</td><td>Fisher 344</td><td>21 </td></tr>\n",
       "\t<tr><td>gse53960</td><td>GSM1328765</td><td>Fisher 344</td><td>21 </td></tr>\n",
       "\t<tr><td>gse53960</td><td>GSM1328766</td><td>Fisher 344</td><td>21 </td></tr>\n",
       "\t<tr><td>gse53960</td><td>GSM1328766</td><td>Fisher 344</td><td>21 </td></tr>\n",
       "\t<tr><td>gse53960</td><td>GSM1328767</td><td>Fisher 344</td><td>21 </td></tr>\n",
       "\t<tr><td>gse53960</td><td>GSM1328767</td><td>Fisher 344</td><td>21 </td></tr>\n",
       "\t<tr><td>gse53960</td><td>GSM1328768</td><td>Fisher 344</td><td>21 </td></tr>\n",
       "\t<tr><td>gse53960</td><td>GSM1328768</td><td>Fisher 344</td><td>21 </td></tr>\n",
       "\t<tr><td>gse53960</td><td>GSM1328769</td><td>Fisher 344</td><td>104</td></tr>\n",
       "\t<tr><td>gse53960</td><td>GSM1328769</td><td>Fisher 344</td><td>104</td></tr>\n",
       "\t<tr><td>gse53960</td><td>GSM1328770</td><td>Fisher 344</td><td>104</td></tr>\n",
       "\t<tr><td>gse53960</td><td>GSM1328770</td><td>Fisher 344</td><td>104</td></tr>\n",
       "\t<tr><td>gse53960</td><td>GSM1328771</td><td>Fisher 344</td><td>104</td></tr>\n",
       "\t<tr><td>gse53960</td><td>GSM1328771</td><td>Fisher 344</td><td>104</td></tr>\n",
       "\t<tr><td>gse53960</td><td>GSM1328772</td><td>Fisher 344</td><td>104</td></tr>\n",
       "\t<tr><td>gse53960</td><td>GSM1328772</td><td>Fisher 344</td><td>104</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 32 × 4\n",
       "\\begin{tabular}{llll}\n",
       " study & sample & strain & age\\_week\\\\\n",
       " <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t gse53960 & GSM1328757 & Fisher 344 & 2  \\\\\n",
       "\t gse53960 & GSM1328757 & Fisher 344 & 2  \\\\\n",
       "\t gse53960 & GSM1328758 & Fisher 344 & 2  \\\\\n",
       "\t gse53960 & GSM1328758 & Fisher 344 & 2  \\\\\n",
       "\t gse53960 & GSM1328759 & Fisher 344 & 2  \\\\\n",
       "\t gse53960 & GSM1328759 & Fisher 344 & 2  \\\\\n",
       "\t gse53960 & GSM1328760 & Fisher 344 & 2  \\\\\n",
       "\t gse53960 & GSM1328760 & Fisher 344 & 2  \\\\\n",
       "\t gse53960 & GSM1328761 & Fisher 344 & 6  \\\\\n",
       "\t gse53960 & GSM1328761 & Fisher 344 & 6  \\\\\n",
       "\t gse53960 & GSM1328762 & Fisher 344 & 6  \\\\\n",
       "\t gse53960 & GSM1328762 & Fisher 344 & 6  \\\\\n",
       "\t gse53960 & GSM1328763 & Fisher 344 & 6  \\\\\n",
       "\t gse53960 & GSM1328763 & Fisher 344 & 6  \\\\\n",
       "\t gse53960 & GSM1328764 & Fisher 344 & 6  \\\\\n",
       "\t gse53960 & GSM1328764 & Fisher 344 & 6  \\\\\n",
       "\t gse53960 & GSM1328765 & Fisher 344 & 21 \\\\\n",
       "\t gse53960 & GSM1328765 & Fisher 344 & 21 \\\\\n",
       "\t gse53960 & GSM1328766 & Fisher 344 & 21 \\\\\n",
       "\t gse53960 & GSM1328766 & Fisher 344 & 21 \\\\\n",
       "\t gse53960 & GSM1328767 & Fisher 344 & 21 \\\\\n",
       "\t gse53960 & GSM1328767 & Fisher 344 & 21 \\\\\n",
       "\t gse53960 & GSM1328768 & Fisher 344 & 21 \\\\\n",
       "\t gse53960 & GSM1328768 & Fisher 344 & 21 \\\\\n",
       "\t gse53960 & GSM1328769 & Fisher 344 & 104\\\\\n",
       "\t gse53960 & GSM1328769 & Fisher 344 & 104\\\\\n",
       "\t gse53960 & GSM1328770 & Fisher 344 & 104\\\\\n",
       "\t gse53960 & GSM1328770 & Fisher 344 & 104\\\\\n",
       "\t gse53960 & GSM1328771 & Fisher 344 & 104\\\\\n",
       "\t gse53960 & GSM1328771 & Fisher 344 & 104\\\\\n",
       "\t gse53960 & GSM1328772 & Fisher 344 & 104\\\\\n",
       "\t gse53960 & GSM1328772 & Fisher 344 & 104\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 32 × 4\n",
       "\n",
       "| study &lt;chr&gt; | sample &lt;chr&gt; | strain &lt;chr&gt; | age_week &lt;chr&gt; |\n",
       "|---|---|---|---|\n",
       "| gse53960 | GSM1328757 | Fisher 344 | 2   |\n",
       "| gse53960 | GSM1328757 | Fisher 344 | 2   |\n",
       "| gse53960 | GSM1328758 | Fisher 344 | 2   |\n",
       "| gse53960 | GSM1328758 | Fisher 344 | 2   |\n",
       "| gse53960 | GSM1328759 | Fisher 344 | 2   |\n",
       "| gse53960 | GSM1328759 | Fisher 344 | 2   |\n",
       "| gse53960 | GSM1328760 | Fisher 344 | 2   |\n",
       "| gse53960 | GSM1328760 | Fisher 344 | 2   |\n",
       "| gse53960 | GSM1328761 | Fisher 344 | 6   |\n",
       "| gse53960 | GSM1328761 | Fisher 344 | 6   |\n",
       "| gse53960 | GSM1328762 | Fisher 344 | 6   |\n",
       "| gse53960 | GSM1328762 | Fisher 344 | 6   |\n",
       "| gse53960 | GSM1328763 | Fisher 344 | 6   |\n",
       "| gse53960 | GSM1328763 | Fisher 344 | 6   |\n",
       "| gse53960 | GSM1328764 | Fisher 344 | 6   |\n",
       "| gse53960 | GSM1328764 | Fisher 344 | 6   |\n",
       "| gse53960 | GSM1328765 | Fisher 344 | 21  |\n",
       "| gse53960 | GSM1328765 | Fisher 344 | 21  |\n",
       "| gse53960 | GSM1328766 | Fisher 344 | 21  |\n",
       "| gse53960 | GSM1328766 | Fisher 344 | 21  |\n",
       "| gse53960 | GSM1328767 | Fisher 344 | 21  |\n",
       "| gse53960 | GSM1328767 | Fisher 344 | 21  |\n",
       "| gse53960 | GSM1328768 | Fisher 344 | 21  |\n",
       "| gse53960 | GSM1328768 | Fisher 344 | 21  |\n",
       "| gse53960 | GSM1328769 | Fisher 344 | 104 |\n",
       "| gse53960 | GSM1328769 | Fisher 344 | 104 |\n",
       "| gse53960 | GSM1328770 | Fisher 344 | 104 |\n",
       "| gse53960 | GSM1328770 | Fisher 344 | 104 |\n",
       "| gse53960 | GSM1328771 | Fisher 344 | 104 |\n",
       "| gse53960 | GSM1328771 | Fisher 344 | 104 |\n",
       "| gse53960 | GSM1328772 | Fisher 344 | 104 |\n",
       "| gse53960 | GSM1328772 | Fisher 344 | 104 |\n",
       "\n"
      ],
      "text/plain": [
       "   study    sample     strain     age_week\n",
       "1  gse53960 GSM1328757 Fisher 344 2       \n",
       "2  gse53960 GSM1328757 Fisher 344 2       \n",
       "3  gse53960 GSM1328758 Fisher 344 2       \n",
       "4  gse53960 GSM1328758 Fisher 344 2       \n",
       "5  gse53960 GSM1328759 Fisher 344 2       \n",
       "6  gse53960 GSM1328759 Fisher 344 2       \n",
       "7  gse53960 GSM1328760 Fisher 344 2       \n",
       "8  gse53960 GSM1328760 Fisher 344 2       \n",
       "9  gse53960 GSM1328761 Fisher 344 6       \n",
       "10 gse53960 GSM1328761 Fisher 344 6       \n",
       "11 gse53960 GSM1328762 Fisher 344 6       \n",
       "12 gse53960 GSM1328762 Fisher 344 6       \n",
       "13 gse53960 GSM1328763 Fisher 344 6       \n",
       "14 gse53960 GSM1328763 Fisher 344 6       \n",
       "15 gse53960 GSM1328764 Fisher 344 6       \n",
       "16 gse53960 GSM1328764 Fisher 344 6       \n",
       "17 gse53960 GSM1328765 Fisher 344 21      \n",
       "18 gse53960 GSM1328765 Fisher 344 21      \n",
       "19 gse53960 GSM1328766 Fisher 344 21      \n",
       "20 gse53960 GSM1328766 Fisher 344 21      \n",
       "21 gse53960 GSM1328767 Fisher 344 21      \n",
       "22 gse53960 GSM1328767 Fisher 344 21      \n",
       "23 gse53960 GSM1328768 Fisher 344 21      \n",
       "24 gse53960 GSM1328768 Fisher 344 21      \n",
       "25 gse53960 GSM1328769 Fisher 344 104     \n",
       "26 gse53960 GSM1328769 Fisher 344 104     \n",
       "27 gse53960 GSM1328770 Fisher 344 104     \n",
       "28 gse53960 GSM1328770 Fisher 344 104     \n",
       "29 gse53960 GSM1328771 Fisher 344 104     \n",
       "30 gse53960 GSM1328771 Fisher 344 104     \n",
       "31 gse53960 GSM1328772 Fisher 344 104     \n",
       "32 gse53960 GSM1328772 Fisher 344 104     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "meta_combine_1<-select(all_meta, 'Sample.Name', 'strain', 'age_week')\n",
    "names(meta_combine_1)[1] <- \"sample\"\n",
    "data.frame(append(meta_combine_1, c(study='gse53960'), after=0))\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "GEO study [GSE162152](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE162152)\n",
    "\n",
    "txt files specific to circRNAs -- need raw data from SRA\n",
    "\n",
    "SRA files:\n",
    "        SRR13142136\n",
    "        SRR13142137\n",
    "        SRR13142138"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "GEO study [GSE85420](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE85420)\n",
    "\n",
    "txt files with processed data:\n",
    "    * mapped to the reference rat genome (rn5) using TopHat 2 aligner\n",
    "    * Aligned reads were used for assembly of novel transcripts and differential expression of novel and reference transcripts with Cuffdiff 2.1.1.\n",
    "    * Supplementary_files_format_and_content: XLSX file showing Log2(FPKM) for control and exposed condition and Log2 of their FPKM ratio for each transcript, as well as transcript coordinates and gene names.\n",
    "\n",
    "\n",
    "**Single - Read length 75** [see here](https://trace.ncbi.nlm.nih.gov/Traces/sra/?run=SRR4017427)\n",
    "\n",
    "\n",
    "SRA files:\n",
    "        SRR4017429\n",
    "        SRR4017448\n",
    "        SRR4017447\n",
    "        SRR4017446\n",
    "        SRR4017445\n",
    "        SRR4017431\n",
    "        SRR4017432\n",
    "        SRR4017433\n",
    "        SRR4017434\n",
    "        SRR4017435\n",
    "        SRR4017436\n",
    "        SRR4017437\n",
    "        SRR4017438\n",
    "        SRR4017439\n",
    "        SRR4017440\n",
    "        SRR4017441\n",
    "        SRR4017442\n",
    "        SRR4017443\n",
    "        SRR4017444"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# load in files from STAR --quantMode genecounts\n",
    "\n",
    "countPATH <- \"/bigdata/faustmanlab/bch/gse85420/counts\"\n",
    "\n",
    "files <- list.files(path=countPATH, pattern=\"*.tab\", full.names=TRUE)\n",
    "names(files) <- lapply(strsplit(files, \"/|_\"), \"[\",7)\n",
    "countslist <- lapply(files, read.delim, sep=\"\\t\", header=FALSE)\n",
    "names(countslist)<- names(files)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "countslist[[1]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "colnames<- c(\"gene\", \"unstranded_counts\", \"htseq_count_Fstreadalign\",\"htseq_countrev_Sndreadalign\")\n",
    "\n",
    "\n",
    "for (i in seq_along(countslist)){\n",
    "  colnames(countslist[[i]]) <- colnames}\n",
    "\n",
    "counts_temp <- lapply(countslist, function(x) {x <- x %>% select(-c(unstranded_counts,htseq_count_Fstreadalign))})\n",
    "counts_use <- lapply(counts_temp, function(x) {x <- x[-c(1:4),]})\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "head(counts_use[[1]])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "all_counts <- reduce(counts_use, full_join, by = \"gene\")\n",
    "colnames(all_counts)<-c(\"gene\", names(counts_use))\n",
    "\n",
    "rownames(all_counts)<- all_counts$gene\n",
    "all_counts <- all_counts[,-1]\n",
    "\n",
    "\n",
    "\n",
    "head(all_counts)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "[GSE41637](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE41637)\n",
    "\n",
    "Supplementary files include mapped reads:\n",
    "    * Mapped to rn4\n",
    "\n",
    "SRA files:\n",
    "        SRR594427 (strain F344/cr1)\n",
    "        SRR594436 (strain BN/SsNHsd)\n",
    "        SRR594445 (strain Sprague-Dawley)\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# To combine datasets into one count and metadata file. Need to standardize metadata entries and add column for study identifier\n",
    "\n",
    "\n",
    "### Target metadata table structure\n",
    "\n",
    "| *Study* | *Strain* | *Sample Name* | *Age (weeks)* |\n",
    "|----|----|----|----|\n",
    "| ... | ... | ... | ... |\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.1.3"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
